tools/twix/src/panels/image_color_select.rs

```diff
@@ -0,0 +1,282 @@
+        })
+    }
+}
+
+impl Widget for &mut ImageColorSelectPanel {
+    fn ui(self, ui: &mut Ui) -> Response {
+        ui.horizontal(|ui| {
+            if self.cycler_selector.ui(ui).changed() {
+                let output = CyclerOutput {
+                    cycler: self.cycler_selector.selected_cycler(),
+                    output: Output::Main {
+                        path: "image".to_string(),
+                    },
+                };
+                self.image_buffer = self.nao.subscribe_output(output);
+            }
+            ui.add(egui::Slider::new(&mut self.brush_size, 1.0..=200.0).text("Brush"));
+        });
+
+        ui.separator();
+
+        let image = match self.get_image() {
+            Ok(image) => image,
+            Err(error) => {
+                return ui.label(format!("{error:#?}"));
+            }
+        };
+
+        let handle = ui
+            .ctx()
+            .load_texture("image", image.clone(), TextureOptions::default())
+            .id();
+        let texture = SizedTexture {
+            id: handle,
+            size: Vec2::new(image.width() as f32, image.height() as f32),
+        };
+        let response = ui.image(texture);
+
+        ui.separator();
+
+        let painter = TwixPainter::<Pixel>::paint_at(ui, response.rect).with_camera(
+            vector![image.width() as f32, image.height() as f32],
+            Similarity2::identity(),
+            CoordinateSystem::LeftHand,
+        );
+
+        if let Some(hoverpos) = response.hover_pos() {
+            let pixel_pos = painter.transform_pixel_to_world(hoverpos);
+            if pixel_pos.x() < image.width() as f32 && pixel_pos.y() < image.height() as f32 {
+                let scroll_delta = ui.input(|input| input.scroll_delta);
+                self.brush_size = (self.brush_size + scroll_delta[1]).clamp(1.0, 200.0);
+
+                let mut max = PixelColor {
+                    red: 0.0,
+                    green: 0.0,
+                    blue: 0.0,
+                };
+                let mut min = PixelColor {
+                    red: 1.0,
+                    green: 1.0,
+                    blue: 1.0,
+                };
+                let mut average = PixelColor {
+                    red: 0.0,
+                    green: 0.0,
+                    blue: 0.0,
+                };
+                let mut pixel_count: usize = 0;
+
+                for i in (pixel_pos.x() as isize - self.brush_size as isize)
+                    ..(pixel_pos.x() as isize + self.brush_size as isize + 1)
+                {
+                    for j in (pixel_pos.y() as isize - self.brush_size as isize)
+                        ..(pixel_pos.y() as isize + self.brush_size as isize + 1)
+                    {
+                        if f32::sqrt(
+                            f32::powi(i as f32 - pixel_pos.x(), 2)
+                                + f32::powi(j as f32 - pixel_pos.y(), 2),
+                        ) <= self.brush_size
```

Comment:
You can and should use iterators and `.filter` or similar functions to achieve this behavior ;) Look at https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter